src/internal/Observable.ts

```diff
@@ -229,7 +229,9 @@ export class Observable<T> implements Subscribable<T> {
       }
     }
 
-    return sink;
+    const subscription = new Subscription();
+    subscription.add(sink);
+    return subscription;
```

Comment:
There are performance implications do us doing this with every subscribe call. I'm not sure this is the *exact* approach we can take. Ideally, we'd figure how where the last subscription was and do it there... but there are other things that are being discussed that will solve this issue.